oauth2client/contrib/gce.py

```diff
@@ -123,3 +127,60 @@ def create_scoped_required(self):
 
     def create_scoped(self, scopes):
         return AppAssertionCredentials(scopes, **self.kwargs)
+
+    def sign_blob(self, blob):
+        """Cryptographically sign a blob (of bytes).
+
+        This method is provided to support a common interface, but
+        the actual key used for a Google Compute Engine service account
+        is not available, so it can't be used to sign content.
+
+        Args:
+            blob: bytes, Message to be signed.
+
+        Raises:
+            NotImplementedError, always.
+        """
+        raise NotImplementedError(
+            'Compute Engine service accounts cannot sign blobs')
+
+    @property
+    def service_account_email(self):
+        """Get the email for the current service account.
+
+        Uses the Google Compute Engine metadata service to retrieve the email
+        of the default service account and then caches it on
+        the current object for future use.
+
+        Returns:
+            string, The email associated with the Google Compute Engine
+            service account.
+        """
+        if self._service_account_email is None:
+            self._service_account_email = _get_service_account_email()
+        return self._service_account_email
+
+
+def _get_service_account_email(http_request=None):
```

Comment:
Please locate this newly-introduced code element earlier in the file than any of its use sites.